src/duke/Ui.java

```diff
@@ -0,0 +1,301 @@
+        }
+        System.out.println("Got it. Duke has marked this task as done:" + "\n" +
+                t.getIndicator() + t.getIcon() + t.name + toPrint + "\n" +
+                "Now you have " + arr.size() + " tasks in the list."
+        );
+    }
+
+    /**
+     * Prints out a string everytime the user deletes a task
+     * @param arr temporary list of tasks
+     * @param t the task to print out
+     */
+    public static void printForDelete(ArrayList<Task> arr, Task t) {
+        String keyword = "";
+        String toPrint = "";
+        if (t instanceof Deadline) {
+            keyword = "by";
+            toPrint = " (" + keyword + ": " + t.time + ")";
+        } else if (t instanceof Event) {
+            keyword = "at";
+            toPrint = " (" + keyword + ": " + t.time + ")";
+        }
+        System.out.println("Got it. Duke has removed this task:" + "\n" +
+                t.getIndicator() + t.getIcon() + t.name + toPrint + "\n" +
+                "Now you have " + arr.size() + " tasks in the list."
+        );
+    }
+
+    /**
+     * Prints out the bye statement
+     */
+    public void respondToBye() {
+        System.out.println("Bye. Please come back soon :(");
+        this.stop();
+    }
+
+    /**
+     * Prints out a string saying that the list is empty
+     */
+    public void respondToEmptyList() {
+        System.out.println("You currently do not have any todos.");
+    }
+
+    /**
+     * Prints out the list of tasks when the user inputs in "list"
+     * @throws Exception
+     */
+    public void respondToList() throws Exception {
+        System.out.println("Here are the tasks in the list: ");
+        s.printList();
+    }
+
+    /**
+     * Prints out a statement when the user forgets to put the searchword.
+     */
+    public void respondToFindWrongSyntax() {
+        System.out.println("Provide a number of the todo that you want to mark as done!");
+    }
+
+    /**
+     * Prints the filtered list.
+     * @param searchWord the word on which the user wants to base their search.
+     * @throws Exception
+     */
+    public void respondToFind(String searchWord) throws Exception {
+        ArrayList<Task> temp = new ArrayList<>();
+        for (Task t: tl.getArr()) {
+            if (t.name.contains(searchWord)) {
+                temp.add(t);
+            }
+        }
+        printForFind(temp);
+    }
+
+    /**
+     * Prints out a string whenever the user inputs in an invalid task number to mark as done
+     */
+    public void respondToDoneFail() {
+        System.out.println("The specified todo does not exist!");
+    }
+
+    /**
+     * Prints out a string whenever the user forgets to input in the task number
+     */
+    public void respondToDoneWrongSyntax() {
+        System.out.println("Provide a number of the todo that you want to mark as done!");
+    }
+
+    /**
+     * Prints out the statement saying that task has been marked as done and updates the list
+     */
```

Comment:
Perhaps here should be a tag `@throws` for the JavaDoc of this method? I observed similar problems with some other methods as well.